pandas/tseries/tests/test_timezones.py

```diff
@@ -533,6 +533,21 @@ def test_ambiguous_nat(self):
         di_test = DatetimeIndex(times, tz='US/Eastern')
         self.assert_numpy_array_equal(di_test, localized)
 
+    def test_nonexistent_raise_coerce(self):
+        from pytz.exceptions import NonExistentTimeError
+        times = ['2015-03-08 01:00', '2015-03-08 02:00', '2015-03-08 03:00']
+        index = DatetimeIndex(times)
+        tz = 'US/Eastern'
+        self.assertRaises(NonExistentTimeError,
+                          index.tz_localize, tz=tz)
+        self.assertRaises(NonExistentTimeError,
+                          index.tz_localize, tz=tz, errors='raise')
+        localized = index.tz_localize(tz=tz, errors='coerce')
+        test_times = ['2015-03-08 01:00-05:00', 'NaT',
+                      '2015-03-08 03:00-04:00']
+        test_localized = DatetimeIndex(test_times)
+        self.assert_numpy_array_equal(localized, test_localized)
```

Comment:
you need to compare it exactly, otherwise it doesn't make sense.

```
In [5]: pd.DatetimeIndex(test_times, dtype='datetime64[ns, US/Eastern]')
Out[5]: DatetimeIndex(['2015-03-08 06:00:00-04:00', 'NaT', '2015-03-08 07:00:00-04:00'], dtype='datetime64[ns, US/Eastern]', freq=None)
```